### PR TITLE
Implement WalletConnectModal

### PR DESCRIPTION
**Pull request by hochiw**
## Features

- Introduced headless ui for the modal component
- Created WalletConnectModal
- Scaffolded WalletProvider to contain the modal
- Introduced AppButton common component

## Screenshots
| Desktop | Mobile |
|----------|-------|
|Screenshot 2022-05-19 at 1 46 04 PM|Screenshot 2022-05-19 at 1 46 12 PM|

based on #40 
refs #32

### CODE REVIEW

File: react-app/src/components/common/Buttons/AppButton.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    "type" | "onClick"
+  > {
+  type: AppButtonType;
+  size: AppButtonSize;
+  messageID: MessageID;
+  onClick?: () => void;
+}
+
+function getButtonTypeClassNames(type: AppButtonType): string {
```
Comment (chihimng) [inline react-app/src/components/common/Buttons/AppButton.tsx:20]: 👍

File: react-app/src/components/WalletConnectModal/WalletConnectModal.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import Divider from "../common/Divider/Divider";
+import { Icon, IconType } from "../common/Icons/Icons";
+import LocalizedText from "../common/Localized/LocalizedText";
+
+interface WalletConnectModalProps {
+  isOpened: boolean;
+  onClose: () => void;
+}
+
+const WalletConnectModal: React.FC<WalletConnectModalProps> = (props) => {
```
Comment (chihimng) [inline react-app/src/components/WalletConnectModal/WalletConnectModal.tsx:17]: Might want to rename this to distinguish from the WalletConnect protocol; like `ConnectWalletModel`?

File: react-app/src/components/WalletConnectModal/WalletConnectModal.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  const onKeplrConnect = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+      // TODO: Handle connection
+    },
+    []
+  );
+
+  const onWalletConnectConnect = useCallback(
```
Comment (chihimng) [inline react-app/src/components/WalletConnectModal/WalletConnectModal.tsx:29]: :sosad:

File: react-app/src/providers/WalletProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+interface UnconnectedWalletContextValue {
+  isConnected: false;
+  openConnectionModal: () => void;
+}
+interface ConnectedWalletContextValue {
+  isConnected: true;
+}
+
+type WalletContextValue =
+  | UnconnectedWalletContextValue
```
Comment (chihimng) [inline react-app/src/providers/WalletProvider.tsx:17]: Disconnected?

**Review by chihimng — CHANGES_REQUESTED** [review]

**Comment by hochiw** [comment]
@chihimng updated :adore2x:

File: react-app/src/components/ConnectWalletModal/ConnectWalletModal.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+                    <p
+                      className={cn(
+                        "flex-1",
+                        "text-sm",
+                        "leading-5",
+                        "font-medium",
+                        "text-likecoin-green"
+                      )}
+                    >
+                      <LocalizedText messageID="WalletConnect.connectModal.keplrWallet" />
```
Comment (chihimng) [inline react-app/src/components/ConnectWalletModal/ConnectWalletModal.tsx:134]: @hochiw can you help rename the strings as well? 🙏

**Review by chihimng — CHANGES_REQUESTED** [review]
otherwise lgtm

File: react-app/src/components/ConnectWalletModal/ConnectWalletModal.tsx (hunk 5 above)
Comment (hochiw) [reply react-app/src/components/ConnectWalletModal/ConnectWalletModal.tsx:134]: oh right

File: react-app/src/providers/WalletProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+type WalletContextValue =
+  | DisconnectedWalletContextValue
+  | ConnectedWalletContextValue;
+
+const WalletContext = React.createContext<WalletContextValue>(null as any);
+
+const WalletProvider: React.FC<WalletProviderProps> = (props) => {
+  const { children } = props;
+
+  const [connectWalletModalActive, setConnectWalletModalActive] =
```
Comment (chihimng) [inline react-app/src/providers/WalletProvider.tsx:25]: `isConnectModalActive`? use `is` prefix for flags?

**Review by chihimng — CHANGES_REQUESTED** [review]

**Comment by hochiw** [comment]
@chihimng updated :adore2x:

**Review by chihimng — APPROVED** [review]

**Pull request merged**